client/components/wcc-settings-form/index.js

```diff
@@ -15,64 +15,84 @@ import ObjectPath from 'objectpath';
 import * as SettingsActions from 'state/settings/actions';
 import * as PackagesActions from 'state/form/packages/actions';
 
-const WCCSettingsForm = ( props ) => {
-	const {
-		layout,
-		saveFormData,
-		formActions,
-		noticeActions,
-	} = props;
-
-	const setIsSaving = ( value ) => formActions.setField( 'isSaving', value );
-	const setSuccess = ( value ) => {
-		formActions.setField( 'success', value );
+const WCCSettingsForm = React.createClass( {
```

Comment:
Good point. So in that case, what do you think about:

1) not implementing this in `WCCSettingsForm` so we can keep it as a functional component
and
2) moving the addEventListener up to client/main.js where we have access to the store and can look at store.getState() to see if pristine is false?